src/main/java/org/folio/service/holdings/HoldingsServiceImpl.java

```diff
@@ -103,13 +103,15 @@ public void waitForCompleteStatus(RMAPITemplateContext context, int retries, Com
   public CompletableFuture<Void> loadHoldings(RMAPITemplateContext context, Integer totalCount, String tenantId) {
 
     CompletableFuture<Void> future = CompletableFuture.completedFuture(null);
+    final Instant updatedAt = Instant.now();
     final int totalRequestCount = getRequestCount(totalCount);
     for (int iteration = 1; iteration < totalRequestCount + 1; iteration++) {
       int finalIteration = iteration;
       future = future
         .thenCompose(o -> context.getLoadingService().loadHoldings(MAX_COUNT, finalIteration))
-        .thenCompose(holding -> saveHolding(holding.getHoldingsList(), tenantId));
+        .thenCompose(holding -> saveHoldings(holding.getHoldingsList(), updatedAt, tenantId));
     }
+    future = future.thenCompose(o -> holdingsRepository.deleteByTimeStamp(updatedAt, tenantId));
```

Comment:
Could you please explain why this is required?

Reply:
sure, @dmtkachenko let's assume we performed first loading and the time stamp of the operation is `2019-06-19T09:28:33.127+00`, so the next scheduled date to load holdings is + 2 days -> `2019-06-21T09:28:33.127+00`. The holdings entries which were removed from RM API between those two days will remain in our table with the time stamp of the previous iteration. This operation `holdingsRepository.deleteByTimeStamp()` removes outdated holding entries.

Reply:
in you sample, updatedAt == 2019-06-21T09:28:33.127+00, correct?

`deleteByTimeStamp()` removes entries from the previous run (records those timestamp is not equal to updatedAt), right?

Reply:
> deleteByTimeStamp() removes entries from the previous run (records those timestamp is not equal to updatedAt), right?

yes